### DISCUSSION TITLE
Cannot allow github actions to create a pull request

### DISCUSSION THREAD

**Discussion by ondrej-langr** [community/community#20435] [Actions]
Hello,

I find myself unable to allow github actions to create a pull request. That option is greyed out like this: 
![image](https://user-images.githubusercontent.com/60289583/178752968-e8a3f39d-0643-41f7-9fa3-d3e98aa86c01.png)

When I select `Read repository contents permission` that option can now be selected, but when I select it and hit save then that option is greyed out again without it being enabled.

I was wondering where have I made a mistake, but I can cant find it.

Would be glad to hear you opinions and resolve this if possible.

**Comment by thuongtruong1009** [comment]
Hi @ondrej-langr 
* This is because Github has a protected branch rule in the setting and you do not allow creating PR. 
* You can change this by proving that you are own pull requests. You can try to create ENV key mode. [Example](https://github.com/Azure/actions-workflow-samples/blob/master/assets/create-secrets-for-GitHub-workflows.md)
* And small example for using a secret key to create PR on your repo. See at [here](https://github.com/thuongtruong1009/a_repo_with_3_stars_and_0_forks/blob/main/.github/workflows/run.yml)

![image](https://user-images.githubusercontent.com/71834167/178764505-49bcc028-1614-4a44-b03e-4be9cf7b0eb9.png)

**Reply by ondrej-langr** [reply]
Hello @thuongtruong1009,

Thank you for your time and effort. but I found solution somewhere else. You are right that I should use provided `GITHUB_TOKEN` (although provided example does not use that) secret that github automatically provides, but I already had this set up (although I did not mention it).

Nonetheless, what I found to be working for me is reading this [here](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-1-passing-the-github_token-as-an-input) with the help of reading what I can set [here](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) and finally stumbling on [this](https://docs.github.com/en/organizations/managing-organization-settings/disabling-or-limiting-github-actions-for-your-organization#setting-the-permissions-of-the-github_token-for-your-organization) where it is stated that I should enable what I was originally trying to enable in my organizations settings...

Interesting enough and I hope this helps someone  🙏🏻
